lms/djangoapps/discussion_api/api.py

```diff
@@ -647,9 +651,11 @@ def update_thread(request, thread_id, update_data):
     # Only save thread object if some of the edited fields are in the thread data, not extra actions
     if set(update_data) - set(actions_form.fields):
         serializer.save()
-        thread_edited.send(sender=None, user=request.user, post=cc_thread)
     api_thread = serializer.data
     _do_extra_actions(api_thread, cc_thread, update_data.keys(), actions_form, context)
+
+    # signal to update Teams information
+    thread_edited.send(sender=None, user=request.user, post=cc_thread)
```

Comment:
Note @peter-fogg: the sent signal in the code in master will not fire for all changes to the thread.  The subsequent call to `_do_extra_actions` may make additional changes (by calling the Comment Client directly).  It is separated out from the other thread fields since additional actions are required.